### ISSUE TITLE
Total unread count not showing until next message arrives 

### ISSUE THREAD

**Issue by pixnbit** [GetStream/stream-chat-android#802]
What would be the equivalent to `Client.shared.rx.unreadCount` on iOS? I notice there's a totalUnreadCount on a user, but it's not reactively observed. Suggestions?

**Comment by JcMinarro** [comment]
Hello @pixnbit 
You need to observer this [livedata](https://github.com/GetStream/stream-chat-android/blob/master/stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/ChatDomain.kt#L46)

```
ChatDomain.instance().totalUnreadCount.observe(lifecycleOwnwer) {
                println("You have $it unread messages")
            }
```

**Comment by pixnbit** [comment]
Ah, I thought everything is accessed by ChatClient.instance(). Good to know!

**Comment by pixnbit** [comment]
But I noticed the initial value not returning, my use case is for badge displaying:

```kotlin
    var count = ChatClient.instance().getCurrentUser()?.totalUnreadCount ?: 0
    badge.isVisible = count > 0
    badge.number = count

    ChatDomain.instance().totalUnreadCount.observe(viewLifecycleOwner) {
      badge.isVisible = it > 0
      badge.number = it
    }
```

Even if I had an intial reading of the value, badge won't show until the next message arrives.

**Comment by JcMinarro** [comment]
Hello @pixnbit 

The `ChatClient` is our "Network API implementation" and `ChatDomain` provide Offline Support and some reactive data where you can subscribe.

I will check why it is not working until a new message arrive. We did some refactoring on our Event Parser and maybe there are some corner-case where it is not updated

**Comment by JcMinarro** [comment]
Hello @pixnbit 
It has been fixed on #816 and it is already integrated into our develop branch. 
It will be included on the following release.
In the meantime, if you would like to test it, you can use the following temporary version:
```
implementation 'com.github.getstream:stream-chat-android:60f0d92900'
```
